### PR TITLE
Add 2.10.0 consolidated release notes

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
Add 2.10.0 consolidated release notes

### Issues Resolved
#3743 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Awaiting highlights.

File: release-notes/opensearch-release-notes-2.10.0.md [hunk 1]
```diff
@@ -0,0 +1,671 @@
+OpenSearch and OpenSearch Dashboards 2.10.0 Release Notes
+
+## Release Highlights
+
+### New Features
+
+### Experimental Features
```
Comment (bbarani) [inline release-notes/opensearch-release-notes-2.10.0.md:7]: @sohami The experimental features related to concurrent search should be added here.

Comment (gaiksaya) [reply release-notes/opensearch-release-notes-2.10.0.md:7]: @jhmcintyre Please let us know when you have release highlights ready. Thank you!

File: release-notes/opensearch-release-notes-2.10.0.md [hunk 2]
```diff
@@ -0,0 +1,687 @@
...
+* Updates for dark mode theme ([#695](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/695))
+* Add default Notification subject  ([#701](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/701))
+
+
+### OpenSearch Common Utils
+* Common utils to support Microsoft teams in notifications ([#428](https://github.com/opensearch-project/common-utils/pull/428))
+* Support list of monitor ids in Chained Monitor Findings ([#514](https://github.com/opensearch-project/common-utils/pull/514))
+
+
+### OpenSearch Custom Codecs
```
Comment (bbarani) [inline release-notes/opensearch-release-notes-2.10.0.md:50]: Is this correct? We released ZSTD in 2.9.0 right ? We moved it to plugin in 2.10.0.

Comment (gaiksaya) [reply release-notes/opensearch-release-notes-2.10.0.md:50]: I believe that is grabbed from here https://github.com/opensearch-project/custom-codecs/blob/main/release-notes/opensearch-custom-codecs.release-notes-2.10.0.0.md

Comment (bbarani) [reply release-notes/opensearch-release-notes-2.10.0.md:50]: Does it makes sense to say "ZSTD has been moved to a plugin" ?

File: release-notes/opensearch-release-notes-2.10.0.md [hunk 3]
```diff
@@ -0,0 +1,687 @@
...
+* Initial release with ZSTD codec support
```
Comment (reta) [inline release-notes/opensearch-release-notes-2.10.0.md:51]:
```suggestion
* Moving ZSTD codec support from OpenSearch core to standalone plugin `custom-codecs`
```

**Review by seanneumann — COMMENTED** [review]
Dashboards content looks great. 👍

**Review by peterzhuamazon — APPROVED** [review]

**Review by bbarani — APPROVED** [review]

**Pull request merged**